src/array.cr

```diff
@@ -1770,4 +1770,15 @@ class Array(T)
       end
     end
   end
+
+  # This method returns the i-th element. If i is equal to ary.size, it returns nil.
+  def bsearch(&block : T -> Int|Float|Bool?)
+    idx=bsearch_index(&block)
+    idx ? self[idx] : nil
```

Comment:
is nil the best thing to return if there is no value.
I like the fetch syntax in Ruby maybe this could raise an error unless there is optional parameter passed.